pywbem/_subscription_manager.py

```diff
@@ -1284,16 +1309,87 @@ def _create_filter(self, server_id, source_namespace, query,
-            filter_path = server.conn.CreateInstance(filter_inst)
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
 
+        # If systemname does not yet exist, create it
+        if server_id not in self._systemnames:
+            self._systemnames[server_id] = filter_inst['SystemName']
+        else:
+            if self._systemnames[server_id] != filter_inst['SystemName']:
+                warnings.warn(
+                    _format("WBEM server may be returning different SystemName "
+                            "property for different operations: received  "
+                            "{0!A} and {0!A}. Owned filters may not be found.",
+                            self._systemnames[server_id],
+                            filter_inst['SystemName']),
+                    UserWarning)
+
         return filter_inst
 
+    def _get_systemname(self, server_id):
+        """
+        Get the SystemName returned for anything returned for this server_id.
+        This method gets the s
+        """
+        if server_id in self._systemnames:
+            return self._systemnames[server_id]
+
+        return None
+
+    def _build_path(self, inst, ns, server_id):
+        """
+        Common method to build a CIM_InstanceName for either filter or
+        destination. Used to build the paths for both destinations and filters.
+        Keybindings will be included only if a system name has already
+        been retrieved from an earlier add_filter or add_destination.
+
+        Parameters:
+          inst :class:`~pywbem.CIMInstance`:
+            Instance of CIM_IndicationFilter or CIM_IndicationHandler
+            from which the CIMInstanceName will be build. The CreationClassName,
+            SystemCreateionClassName property values from inst will be used
+            along with the systemname (if already retrieved) and Name
+            property to build the CIM_InstanceName.
+
+          ns :term:`string`:
+            string containing the namespace to be included in the
+            path
+
+          server_id (:term:`string`):
+            The server ID of the WBEM server, returned by
+            :meth:`~pywbem.WBEMSubscriptionManager.add_server`.
+        Returns:
+            CIMInstanceName with path including keybindings or None if
+            there is systemname defined for this server_id
+
+        Returns:
+          CIMInstanceName for inst
+        """
+        svr_name = self._get_systemname(server_id)
+
+        # if svr_name exists, build keybindings, else empty keybindings
+        if svr_name:
+            kb = [('CreationClassName', inst.classname),
+                  ('SystemCreationClassName', SYSTEM_CREATION_CLASSNAME),
+                  ('SystemName', svr_name),
+                  ('Name', inst['Name'])]
+        else:
+            kb = {}
```

Comment:
If `_get_systemname()` is changed as suggested, its return value will never be `None`, and the else path is not needed.

Reply:
DONE: